firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -119,6 +120,56 @@ def __ne__(self, other):
             return not equality_val
 
 
+class FieldPath(object):
+    """ Field Path object for client use.
+
+    Args:
+        parts (string or iterable): Indicating path of the key to be used.
+    """
+
+    def __init__(self, *parts):
+        invalid_characters = '~*/[]'
+        if len(parts) == 1:
+            parts = parts[0]
+        if isinstance(parts, basestring):
+            parts = tuple(parts.split('.'))
+        for part in parts:
+            if (not part
+                or not isinstance(part, basestring)
+                or part in invalid_characters):
+                raise ValueError
+        self.parts = tuple(parts)
+
+    def to_api_repr(self):
+        """ Returns string representation of the FieldPath
+
+        Returns: string representation of the path stored within
+        """
+        pattern = re.compile(r'[A-Za-z_][A-Za-z_0-9]*')
+        ans = []
+        for part in self.parts:
+            match = re.match(pattern, part)
+            if match:
+                ans.append(part)
+            else:
+                part_ans = ''
```

Comment:
This is quadratic. Maybe you can do something with replace. See the [node implementation](https://github.com/googleapis/nodejs-firestore/blob/master/src/path.js#L520).